tests/client.rs

```diff
@@ -1477,7 +1475,7 @@ fn client_hover_after_src_line_change() {
     let contents = ["&str", "let world = \"world\";"];
     let contents: Vec<_> = contents
         .iter()
-        .map(|value| LanguageString { language: "rust".to_string(), value: value.to_string() })
+        .map(|value| LanguageString { language: "rust".to_string(), value: (*value).to_string() })
```

Comment:
Could we use `ToOwned::to_owned` in both cases?

Reply:
The idea is that `(*a).to_string()` is faster than `a.to_string()` for `&&str`
https://rust-lang.github.io/rust-clippy/master/index.html#inefficient_to_string
````
warning: calling `to_string` on `&&str`
    --> tests/client.rs:1480:76
     |
1480 |         .map(|value| LanguageString { language: "rust".to_string(), value: value.to_string() })
     |                                                                            ^^^^^^^^^^^^^^^^^ help: try dereferencing the receiver: `(*value).to_string()`
     |
     = note: `#[warn(clippy::inefficient_to_string)]` on by default
     = help: `&str` implements `ToString` through a slower blanket impl, but `str` has a fast specialization of `ToString`
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#inefficient_to_string